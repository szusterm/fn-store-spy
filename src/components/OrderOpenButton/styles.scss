@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '../../scss/shadow';
@import '../../scss/transition';
@import '../../scss/opacity-colors';
@import '../../scss/input-reset';

.order-open-button {
	$size: 5.5em;
	$icon-size: 1.8em;

	@include input-reset();

	cursor: pointer;

	z-index: 3;
	position: fixed;
	right: 20vh;
	bottom: 20vh;

	width: $size;
	height: $size;
	border-radius: 50%;
	color: #43A047;

	display: flex;
	justify-content: center;
	align-items: center;

	@include white-background(1);
	@include shadow(sm);
	@include transition(md);
	transition-property: box-shadow, transform;

	@include media-breakpoint-down(md) {
		right: 50%;
		bottom: 10vh;
	}

	&:hover {
		@include shadow(md);
	}

	transform: translate(50%, 50%) scale(1);

	&--closed {
		transform: translate(50%, 50%) scale(0);
	}

	&--icon {
		font-size: $icon-size;
	}
}