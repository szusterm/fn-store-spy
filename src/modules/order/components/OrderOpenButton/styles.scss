@import '~material-colors/dist/colors';

@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '~scss/_shadow';
@import '~scss/_transition';
@import '~scss/_opacity-colors';
@import '~scss/_input';
@import '~scss/_center-items';

.order-open-button {
	$size: 5.5em;
	$icon-size: 1.8em;

	@include input-reset();

	cursor: pointer;

	z-index: 3;
	position: fixed;
	right: 20vh;
	bottom: 20vh;

	width: $size;
	height: $size;
	border-radius: 50%;
	color: $md-green-500;

	@include center-items();
	@include white-background(1);
	@include shadow(sm);
	@include transition(md);
	transition-property: box-shadow, transform;

	@include media-breakpoint-down(md) {
		right: 50%;
		bottom: 10vh;
	}

	&:hover {
		@include shadow(md);
		color: $md-green-600;
	}

	transform: translate(50%, 50%) scale(1);

	&--hidden {
		transform: translate(50%, 50%) scale(0);
	}

	&--icon {
		font-size: $icon-size;
	}

	&--items-count {
		position: absolute;
		top: 0;
		right: 0;
	}
}