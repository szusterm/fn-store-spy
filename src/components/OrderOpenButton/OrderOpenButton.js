import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {openOrderList} from '../../redux/actions';

import './styles.scss';

export class OrderOpenButton extends Component {
	checkThatOrderIsEmpty() {
		const {orderedItems} = this.props;
		return (orderedItems.length === 0);
	}

	render() {
		return (
			<button
				className={`order-open-button ${(this.checkThatOrderIsEmpty()) && 'order-open-button--closed'}`}
				onClick={this.props.openOrderList}
			>
				<div className={'material-icons'}>
					<span className={'order-open-button--icon'}>shopping_basket</span>
				</div>
			</button>
		);
	}
}

OrderOpenButton.propTypes = {
	orderedItems: PropTypes.array, //redux
	listOpened: PropTypes.bool, //redux
	openOrderList: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {listOpened, items} = state.order;
	return {
		orderedItems: items,
		listOpened
	};
};

const mapDispatchToProps = {openOrderList};

export default connect(mapStateToProps, mapDispatchToProps)(OrderOpenButton);