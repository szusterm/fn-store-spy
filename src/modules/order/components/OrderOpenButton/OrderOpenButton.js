import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {openOrderList} from '../../../../redux/actions';

import CircleOrderCount from '../CircleOrderCount/index';

import './styles.scss';

export class OrderOpenButton extends Component {
	checkThatOrderIsEmpty() {
		const {orderedItems} = this.props;
		return (orderedItems.length === 0);
	}

	render() {
		return (
			<button
				className={`order-open-button ${(this.checkThatOrderIsEmpty()) && 'order-open-button--hidden'}`}
				onClick={this.props.openOrderList}
			>
				<div className={'material-icons'}>
					<span className={'order-open-button--icon'}>shopping_basket</span>
				</div>
				<div className={'order-open-button--items-count'}>
					<CircleOrderCount/>
				</div>
			</button>
		);
	}
}

OrderOpenButton.propTypes = {
	orderedItems: PropTypes.array, //redux
	openOrderList: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items} = state.order;
	return {
		orderedItems: items,
	};
};

const mapDispatchToProps = {openOrderList};

export default connect(mapStateToProps, mapDispatchToProps)(OrderOpenButton);