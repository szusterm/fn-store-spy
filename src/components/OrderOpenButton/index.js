import OrderOpenButton from './OrderOpenButton';

export default OrderOpenButton;