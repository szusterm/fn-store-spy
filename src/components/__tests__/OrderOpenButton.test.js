import React from 'react';
import {shallow} from 'enzyme';

import {OrderOpenButton} from '../OrderOpenButton/OrderOpenButton';

const setup = (propOverrides) => {
	const props = Object.assign({
		orderedItems: [], //redux
		openOrderList: () => true //redux
	}, propOverrides);

	const wrapper = shallow(<OrderOpenButton {...props}/>);

	return {props, wrapper};
};

describe('CircleOrderCount Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('hides a button, if order is empty', () => {
		const {wrapper} = setup();

		const hiddenButtonClass = '.order-open-button--closed';

		wrapper.setProps({orderedItems: ['item0']});
		expect(wrapper.find(hiddenButtonClass).exists()).toBe(false);

		wrapper.setProps({orderedItems: []});
		expect(wrapper.find(hiddenButtonClass).exists()).toBe(true);
	});
});