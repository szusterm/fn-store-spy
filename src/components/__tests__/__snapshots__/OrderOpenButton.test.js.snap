// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleOrderCount Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderOpenButton
    openOrderList={[Function]}
    orderedItems={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="material-icons"
        >
          <span
            className="order-open-button--icon"
          >
            shopping_basket
          </span>
        </div>,
        <div
          className="order-open-button--items-count"
        >
          <Connect(CircleOrderCount) />
        </div>,
      ],
      "className": "order-open-button order-open-button--closed",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="order-open-button--icon"
          >
            shopping_basket
          </span>,
          "className": "material-icons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "shopping_basket",
            "className": "order-open-button--icon",
          },
          "ref": null,
          "rendered": "shopping_basket",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(CircleOrderCount) />,
          "className": "order-open-button--items-count",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="material-icons"
          >
            <span
              className="order-open-button--icon"
            >
              shopping_basket
            </span>
          </div>,
          <div
            className="order-open-button--items-count"
          >
            <Connect(CircleOrderCount) />
          </div>,
        ],
        "className": "order-open-button order-open-button--closed",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="order-open-button--icon"
            >
              shopping_basket
            </span>,
            "className": "material-icons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "shopping_basket",
              "className": "order-open-button--icon",
            },
            "ref": null,
            "rendered": "shopping_basket",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(CircleOrderCount) />,
            "className": "order-open-button--items-count",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
